gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/AllowedAccountsSupport.java

```diff
@@ -0,0 +1,47 @@
+package com.netflix.spinnaker.gate.security;
+
+import com.netflix.spinnaker.fiat.model.Authorization;
+import com.netflix.spinnaker.fiat.model.resources.Account;
+import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator;
+import com.netflix.spinnaker.gate.services.CredentialsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Returns list of WRITE enabled accounts to populate X-SPINNAKER-ACCOUNTS header.
```

Comment:
Not sure why - With Fiat, the `X-SPINNAKER-ACCOUNTS` header is not used.

Reply:
the purpose of supplying it is to allow a migration. we need authenticated sessions in fiat (so gate runs with fiat.enabled) before we can turn on fiat in clouddriver/front50 otherwise all the auth checks in those services will fail. But turning fiat.enabled on in gate would stop propagating the accounts header